src/disk.hpp

```diff
@@ -201,6 +206,9 @@ struct FileDisk {
             }
             amtwritten =
                 ::fwrite(reinterpret_cast<const char *>(memcache), sizeof(uint8_t), length, f_);
+#ifdef _WIN32
+            _commit(_fileno(f_));
```

Comment:
it's not obvious that this is an improvement. This will block until the bytes have made it to the disk, right?
Could you provide some explanation of the symptom you're experiencing that this solves?
Ideally it would be explained in a comment here as well, to let future readers of the code understand why this call is here

Reply:
The symptom is seen in https://github.com/Chia-Network/chia-blockchain/issues/2337

Reply:
could you provide measurements demonstrating the improvement with this patch?

Reply:
the reason I feel cautious about this is that you really want disk *writes* to happen in parallel with your program executing, which means sticking the data in kernel buffers that are flushed to disk at some later time, while the program keeps running. As far as I can tell, this will effectively stop the program to wait for the data to make it all the way down to the platter (presumably surviving a power outage).

Generally, you save a lot of time to keep running instead.

Reply:
the [documentation](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/commit?view=msvc-160) doesn't actually say whether the call blocks until the data has been flushed or not.